src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/mappers/TransferIndexGenerator.java

```diff
@@ -58,4 +72,18 @@ private void generateTransfers(TripPatternWithRaptorStopIndexes pattern) {
             }
         }
     }
+
+    private void generateForbiddenTransfers(
+        TripPatternWithRaptorStopIndexes pattern
+    ) {
+        int nStops = pattern.getPattern().getStops().size();
+        List<Stop> stops = pattern.getPattern().getStops();
+        for (int stopPos=0; stopPos < nStops; ++stopPos) {
+            var forbiddenTransfers = transferService.listForbiddenTransfersTo(stops.get(stopPos));
+            for (Transfer tx : forbiddenTransfers) {
+                // TODO: Do we need to handle transfersFrom?
```

Comment:
I'm not sure if its needed, but for reverse search should we also add to forbiddenTransferFrom?
If this is the case, is
```java
private void generateForbiddenTransfers(
        TripPatternWithRaptorStopIndexes pattern
    ) {
        int nStops = pattern.getPattern().getStops().size();
        List<Stop> stops = pattern.getPattern().getStops();
        for (int stopPos=0; stopPos < nStops; ++stopPos) {
            var toForbiddenTransfers = transferService.listForbiddenTransfersTo(stops.get(stopPos));
            for (Transfer tx : toForbiddenTransfers) {
                pattern.addForbiddenTransfersTo(tx, stopPos);
            }
            var fromForbiddenTransfers =
                transferService.listForbiddenTransfersFrom(stops.get(stopPos));
            for (Transfer tx : fromForbiddenTransfers) {
                pattern.addForbiddenTransfersFrom(tx, stopPos);
            }
        }
    }
```
the right way to do it?

Reply:
I have not looked at your latest changes yet, but we plan to make the reverse search work for the McRaptor in the near feature to compute a better Heuristic - in that case it would be nice to also include this. I think this need to be accounted for in the Heuristics for it to work correctly.